clang/lib/CConv/RewriteUtils.cpp

```diff
@@ -1322,10 +1338,51 @@ class CastPlacementVisitor :
       Writer.InsertTextBefore(E->getBeginLoc(), CastPrefix);
     }
   }
+
+  bool needsFancyCast(std::set<ConstraintVariable *> &LHSCons, Expr *E,
+                      std::string &CastStr) {
+    ConstraintVariable *LCVariable = nullptr;
+    bool LHSChkType = false;
+    bool NeedFancyCast = false;
+    auto &CS = Info.getConstraints();
+    for (auto *LC : LHSCons) {
+      if (LC->anyChanges(CS.getVariables())) {
+        LCVariable = LC;
+        LHSChkType = true;
+        break;
```

Comment:
Why do you break out immediately? Is it possible there are multiple LHSs in which case you'll not do them all? If you think it's not possible, use `Util::getOnly`